File: workbook/백트래킹/15649.py
import sys
input = sys.stdin.readline

def recur(cnt, path):
    if cnt >= m:
        print(*path)
        return

    for i in range(1, n+1):
        if visit[i] == 0:
            visit[i] = 1
            recur(cnt + 1, path + [i])
            visit[i] = 0

n, m = map(int, input().split())
visit = [0 for _ in range(n + 1)]
recur(0, [])

File: workbook/백트래킹/15649.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString();
// let input = fs.readFileSync("ex.txt").toString();

const [n, m] = input.split(" ");
const visit = new Array(parseInt(n) + 1).fill(0);

function recur(cnt, path) {
  if (cnt >= m) {
    console.log(path.join(" "));
    return;
  }
  for (let i = 1; i < n + 1; i++) {
    if (visit[i] === 0) {
      visit[i] = 1;
      recur(cnt + 1, path.concat(i));
      visit[i] = 0;
    }
  }
}
recur(0, []);
